saleor/payment/utils.py

```diff
@@ -433,3 +433,29 @@ def price_to_minor_unit(value: Decimal, currency: str):
     number_places = Decimal("10.0") ** precision
     value_without_comma = value * number_places
     return str(value_without_comma.quantize(Decimal("1")))
+
+
+def get_channel_slug_from_payment(payment: Payment) -> Union[str, None]:
+    if payment.checkout:
+        channel_slug = payment.checkout.channel.slug
+    elif payment.order:
+        channel_slug = payment.order.channel.slug
+    else:
+        channel_slug = None  # type: ignore
+
+    return channel_slug
+
+
+def try_void_or_refund_inactive_payment(
+    payment: Payment, transaction: Transaction, manager: "PluginsManager"
+):
+    """Handle refund or void inactive payments.
+
+    In case when we have open multiple payments for single checkout but only one is
+    active. Some payment methods don't required confirmation so we can receive delayed
+    webhook when we have order already paid.
+    """
+    if transaction.is_success:
+        update_payment_charge_status(payment, transaction)
+        channel_slug = get_channel_slug_from_payment(payment)
```

Comment:
I see that channel_slug is available in `handle_authorized_payment_intent` as an argument, so maybe we don't have to fetch it again? Maybe we could receive the channel easily in other places as well? I would check if we can get rid of extra DB queries here.

Reply:
It's a common utility for Adyen and Stripe and with Stripe it will work like a charm but with Adyen not really because we don't receive channel in handlers. Only way to get channel is get it from payment.order or payment.checkout. It's a good point overall because less queries is better but in this case I will need separate functions...